src/devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1114 @@
+            catch (IOException ex)
+            {
+
+                throw new IOException($"Can't change I2C Address to {newAddress}", ex);
+            }
+        }
+
+        /// <summary>
+        /// Start continuous ranging measurements. If periodMilliseconds is 0
+        /// continuous back-to-back mode is used (the sensor takes measurements as
+        /// often as possible) otherwise, continuous timed mode is used, with the given
+        /// inter-measurement period in milliseconds determining how often the sensor
+        /// takes a measurement.
+        /// </summary>
+        /// <param name="periodMilliseconds">The interval period between 2 measurements. Default is 0</param>
+        public void StartContinuousMeasurement(Int32 periodMilliseconds = 0)
+        {
+            // Initialize the measurement
+            InitMeasurement();
+            _continuousInitialized = true;
+            if (periodMilliseconds != 0)
+            {
+                // If we have a period, then change the register for continuous measurement
+                var osc_calibrate_val = ReadUInt16((byte)Registers.OSC_CALIBRATE_VAL);
+                if (osc_calibrate_val != 0)
+                    periodMilliseconds *= osc_calibrate_val;
+                WriteUInt32((byte)Registers.SYSTEM_INTERMEASUREMENT_PERIOD, (UInt32)periodMilliseconds);
+                WriteRegister((byte)Registers.SYSRANGE_START, 0x04);
+            }
+            else
+            {
+                // If we don't just start the continuous measurement
+                WriteRegister((byte)Registers.SYSRANGE_START, 0x02);
+            }
+        }
+
+        /// <summary>
+        /// Reads the measurement when the mode is set to continuious. 
+        /// </summary>
+        /// <returns>The range in millimeters, a maximum value is returned depending on the various settings</returns>
+        private UInt16 ReadContinuousMeasrurementMillimeters()
+        {
+            Stopwatch stopWatch = Stopwatch.StartNew();
+            var expirationMilliseconds = stopWatch.ElapsedMilliseconds + _operationTimeout;
+            while ((ReadByte((byte)Registers.RESULT_INTERRUPT_STATUS) & 0x07) == 0)
+            {
+                if (stopWatch.ElapsedMilliseconds > expirationMilliseconds)
+                    throw new IOException($"{nameof(ReadContinuousMeasrurementMillimeters)} timeout error");
+            }
+            // assumptions: Linearity Corrective Gain is 1000 (default)
+            // fractional ranging is not enabled
+            var range = ReadUInt16((byte)Registers.RESULT_RANGE_STATUS + 10);
+            WriteRegister((byte)Registers.SYSTEM_INTERRUPT_CLEAR, 0x01);
+            return range;
+        }
+
+        /// <summary>
+        /// Get a distance in millimeters from the continous measurement feature.
+        /// It is recommended to used this method to gethigher quality measurements
+        /// </summary>
+        public UInt16 DistanceContinousMillimeters
```

Comment:
I think you can add an enum type `MeasuringMode`, change the measurement by setting properity, rename `DistanceContinousMillimeters` to `Distance`.

Reply:
Yes, good idea, will do it like that. I was hesitating but it will be much simpler.